src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp

```diff
@@ -262,155 +356,200 @@ class boss_paletress : public CreatureScript
 public:
     boss_paletress() : CreatureScript("boss_paletress") { }
 
-    struct boss_paletressAI : public ScriptedAI
+    struct boss_paletressAI : public BossAI
     {
-        boss_paletressAI(Creature* creature) : ScriptedAI(creature)
+        boss_paletressAI(Creature* creature) : BossAI(creature, DATA_ARGENT_CONFESSOR_PALETRESS)
         {
             Initialize();
-            instance = creature->GetInstanceScript();
 
-            creature->SetReactState(REACT_PASSIVE);
-            creature->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE);
-            creature->RestoreFaction();
+            me->SetReactState(REACT_DEFENSIVE);
         }
 
         void Initialize()
         {
-            uiHolyFireTimer = urand(9000, 12000);
-            uiHolySmiteTimer = urand(5000, 7000);
-            uiRenewTimer = urand(2000, 5000);
-
-            uiResetTimer = 7000;
-
-            bHealth = false;
-            bDone = false;
+            bMemory = false;
         }
 
-        InstanceScript* instance;
         ObjectGuid MemoryGUID;
-
-        bool bHealth;
-        bool bDone;
-
-        uint32 uiHolyFireTimer;
-        uint32 uiHolySmiteTimer;
-        uint32 uiRenewTimer;
-        uint32 uiResetTimer;
+        bool bMemory;
 
         void Reset() override
         {
-            me->RemoveAllAuras();
-
-            Initialize();
-
-            if (Creature* pMemory = ObjectAccessor::GetCreature(*me, MemoryGUID))
-                if (pMemory->IsAlive())
-                    pMemory->RemoveFromWorld();
+            if (instance->GetBossState(DATA_ARGENT_CONFESSOR_PALETRESS) != SPECIAL)
+            {
+                me->RemoveAura(SPELL_SHIELD);
+                Initialize();
+                _Reset();
+            }
         }
 
         void SetData(uint32 uiId, uint32 /*uiValue*/) override
         {
             if (uiId == 1)
+            {
+                me->InterruptNonMeleeSpells(true);
+                Talk(SAY_MEMORY_DEATH);
+                me->GetMotionMaster()->Clear();
                 me->RemoveAura(SPELL_SHIELD);
+            }
         }
 
         void DamageTaken(Unit* /*done_by*/, uint32 &damage) override
         {
+            if (!bMemory && !HealthAbovePct(25))
+            {
+                me->InterruptNonMeleeSpells(true);
+                DoCastAOE(SPELL_HOLY_NOVA, false);
```

Comment:
mean i should delete this line "DoCastAOE(SPELL_HOLY_NOVA, false);"
@Keader thanks for your guid